libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -491,14 +510,18 @@ void YamlConfiguration::parsePortYaml(YAML::Node *portNode,
 
   YAML::Node inputPortsObj = portNode->as<YAML::Node>();
 
-  // generate the random UIID
-  uuid_generate(uuid);
-
-  auto portId = inputPortsObj["id"].as<std::string>();
+  // Check for required fields
+  checkRequiredField(&inputPortsObj, "name");
   auto nameStr = inputPortsObj["name"].as<std::string>();
+  checkRequiredField(&inputPortsObj, "id", "The field 'id' is required for "
+      "the port named '" + nameStr + "' in the YAML Config. If this port "
+      "is specificy an input port for a NiFi Remote Process Group, the port "
+      "id should match the id of of the input port in the NiFi configuration. "
+      "This is a UUID of the format XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX.");
+  auto portId = inputPortsObj["id"].as<std::string>();
   uuid_parse(portId.c_str(), uuid);
 
-  port = new minifi::RemoteProcessorGroupPort(nameStr.c_str(), uuid);
+  port = new minifi::RemoteProcessorGroupPort(nameStr, uuid);
 
   processor = (std::shared_ptr<core::Processor>) port;
```

Comment:
This will create a true shared pointer object for port and perform a compile time cast. I think the only reason port exists is to call RPG specific functions later, which predates me.